src/fides/api/ops/service/connectors/saas_query_config.py

```diff
@@ -218,16 +267,17 @@ def _generate_product_list(*args: Dict[str, Any]) -> List[Dict[str, Any]]:
         ]
 
     def generate_query(
-        self, input_data: Dict[str, List[Any]], policy: Optional[Policy]
+        self,
+        input_data: Dict[str, List[Any]],
+        policy: Optional[Policy],
     ) -> SaaSRequestParams:
         """
         This returns the method, path, header, query, and body params needed to make an API call.
         This is the API equivalent of building the components of a database
         query statement (select statement, where clause, limit, offset, etc.)
         """
 
-        current_request: Optional[SaaSRequest] = self.get_request_by_action("read")
```

Comment:
We set the `self.current_state` so we don't break the method signature for `generate_query`